### PR TITLE
Milestone two

### PR DESCRIPTION
**Pull request by expjazz**
In this milestone, we have created:
 - Components directory with the required components
 - Containers directory
 - Actions directory with all the required actions
 - Reducers directory with 2 files: books.js & index.js
- We used the UUID instead of math.random for the book id

### We are using hooks in this project. Kindly let us know if that is ok since in the requirements it asks to use class-based components. Since the react documentation is shifting to hooks, that's what we did. You can see that all the functionalities will be the same. Instead of using the _connect and map state_ to props, we are applying the useSelector.

### CODE REVIEW

File: src/components/BooksForm.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+import React, { useState } from 'react';
```
Comment (bellom) [inline src/components/BooksForm.js]: Hi Guys, Good job so far.

Let's do some rearrangement.

Kindly move this file  into `src/container` dir

React bindings for Redux separate presentational components from container components. This approach can make your app easier to understand and allow you to more easily reuse components. In your components directory (presentational components) you have components that are not connected to the Redux store, and on your containers directory you have components that are connected to the Redux store. https://redux.js.org/basics/usage-with-react#presentational-and-container-components 

image
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896827/

File: src/components/Book.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      <td>
+
+        { category}
+      </td>
+    </StyledRow>
+  );
+}
+
+Book.propTypes = {
+  // eslint-disable-next-line react/forbid-prop-types
```
Comment (bellom) [inline src/components/Book.js]: using PropTypes.shape here will allow you set a type check that is more accurate.

File: src/components/Book.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+const StyledRow = styled.tr`
+  background: purple;
+  color: yellow;
+`;
+export default function Book(props) {
```
Comment (bellom) [inline src/components/Book.js]: In here; I think you can simplify by using arrow function like this 👇 right?
`const Book = ({ book: { title, category } }) => (
`
then make use of PropTypes.shape to capture all the props.

File: src/App.js [hunk 4]
```diff
@@ -1,8 +1,14 @@
 import React from 'react';
-import './App.css';
+import Bookslist from './components/BooksList';
```
Comment (bellom) [inline src/App.js]: Let's do some rearrangement in here.

Kindly move `app.js` into `src/component` dir

image
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896827/

File: src/components/BooksList.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
```
Comment (bellom) [inline src/components/BooksList.js]: Kindly move this file  into `src/container` dir

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi Team :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 
 
**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

File: src/components/BooksForm.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+      title: '',
+    });
+  };
+
+  const handleChange = e => {
+    setFormInputs({
+      ...formInputs,
+      title: e.target.value,
+    });
+  };
```
Comment (bellom) [inline src/components/BooksForm.js]: I will advise you to work on the milestones one after another(or write inside another branch), `handleSubmit` and `handleChange` features should be in milestone 3.

File: src/actions/index.js [hunk 7]
```diff
@@ -0,0 +1,3 @@
+const addBook = (content = '') => ({ type: 'CREATE_BOOK', payload: content });
+
+export default addBook;
```
Comment (shshamim63) [inline src/actions/index.js]: An action regarding to REMOVE_BOOK is missing in here. 
image
Please add it.

**Review by shshamim63 — CHANGES_REQUESTED** [review]
## :recycle: Required Changes :recycle: 
___
Hi @expjazz and @mariosknl :wave:,
Almost completed this milestone, just found two issues and these are very easy to resolve. Let's fix them.
___
### :spades: Mandatory Changes :spades: 
___
- Your code should not contain any unnecessary files. But inside the component directory, there is an empty index file. please remove it.
- You forgot to add another method in the actions/index.js. Please check the comment below.
___
#### Please resolve the mentioned issue and send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

**Comment by expjazz** [comment]
- We removed the empty files and added the method that was missing

**Review by fatymahmed — APPROVED** [review]
## Hi @expjazz @mariosknl   
### Good job making the requested changes 🎉🎉
## Status: Approved ✔️

**Pull request closed without merging**